code/game/objects/items/chainsaw.dm

```diff
@@ -78,6 +78,40 @@
 		var/datum/action/A = X
 		A.UpdateButtonIcon()
 
+/obj/item/chainsaw/afterattack(atom/A, mob/user, proximity)
+	if(!on)
+		return ..()
+
+	var/lignocellulose = counterlist_sum(A.has_material_type(/datum/material/wood))
+
+	if(!lignocellulose)
+		return ..()
+
+
+
+	var/dust_harvest = lignocellulose * 0.1 //you lose 10% as sawdust
+	lignocellulose *= 0.9
+	dust_harvest = round((lignocellulose % MINERAL_MATERIAL_AMOUNT) / 100) //the leftover wood material also becomes sawdust.
+	var/plank_harvest = round(lignocellulose / (MINERAL_MATERIAL_AMOUNT))
+
+	if(plank_harvest >= 3)
+		user.visible_message("<span class='notice'>[user] starts sawing [A] to pieces!</span>", "<span class='notice'>You start sawing [A] to pieces!</span>")
+		if(!do_after(user, 2 SECONDS, target = A))
+			return
+
+	if(plank_harvest)
+		new /obj/item/stack/sheet/mineral/wood(get_turf(A),plank_harvest) //spawn an amount of planks equal to plank_harvest
+
+	if(lignocellulose % MINERAL_MATERIAL_AMOUNT)
```

Comment:
I'm confused, what is this supposed to do? I know what modulo does, but am not sure why it's here, especially since the code below suggests that this should just be if there's any harvest at all.

Reply:
Good catch!

I changed it toif(dust_amount) to prevent it from spawning empty decals incase something goes wrong.